File: Golang/Minimum Value to Get Positive Step by Step Sum.go
func minStartValue(nums []int) int {
    currSum := 0
    minSum := 0
    for _, num := range nums {
        currSum = currSum + num
        minSum = min(minSum, currSum)
    }
    if minSum > 0 {
        return 1
    } else {
        return 1 - minSum
    }
}

func min(a int, b int) int {
    if a < b {
        return a
    }
    return b
}

File: Python3/Minimum Value to Get Positive Step by Step Sum.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        cur_sum , min_sum = 0, 0

        for num in nums:
            cur_sum = cur_sum + num
            min_sum = min(min_sum, cur_sum)
        
        if min_sum > 0:
            return 1
        else:
            return 1 - min_sum
        